firrtl/src/main/scala/firrtl/annotations/Target.scala

```diff
@@ -139,11 +133,10 @@ object Target {
   def deserialize(s: String): Target = {
     val regex = """(?=[~|>/:.\[])"""
     s.split(regex)
-      .foldLeft(GenericTarget(None, None, Vector.empty)) { (t, tokenString) =>
+      .foldLeft(GenericTarget(None, Vector.empty)) { (t, tokenString) =>
         val value = tokenString.tail
         tokenString(0) match {
-          case '~' if t.circuitOpt.isEmpty && t.moduleOpt.isEmpty && t.tokens.isEmpty =>
-            if (value == "???") t else t.copy(circuitOpt = Some(value))
+          case '~' if t.moduleOpt.isEmpty && t.tokens.isEmpty => t
```

Comment:
Note: the effect of this is that this will correctly tokenize targets which both have and do not have circuits. If there is a circuit, it will just be discarded. This is why the Trace API continues to work even though CIRCT will produce annotations that include a circuit.